### DISCUSSION TITLE
syntax error when parsing year type in mysql

### DISCUSSION THREAD

**Discussion by zhengshuoo** [sqldef/sqldef#634] [Feature requests]
## Platform
https://sqldef.github.io/

### Current schema
```
CREATE TABLE `test` (
  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT COMMENT '主键',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COMMENT='test';
```
### New schema
```
CREATE TABLE `test` (
  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT COMMENT '主键',
  `year_field` year(4) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COMMENT='test';
```
## Current output
```
found syntax error when parsing DDL "CREATE TABLE `test` (
  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT COMMENT '主键',
  `year_field` year(4) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COMMENT='test'": syntax error at position 112
```
## Expected output
```
CREATE TABLE `test` (
  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT COMMENT '主键',
  `year_field` year(4) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COMMENT='test';
```

**Comment by AlexOS888** [comment]
Hello, is there any update on this incident?

**Comment by okaponta** [comment]
As shown in the following document, YEAR(4) has been deprecated, and it is recommended to use YEAR instead. 
https://dev.mysql.com/doc/refman/9.4/en/year.html

Since using YEAR does not cause errors and the official documentation states that YEAR(4) will be removed in the future, I think it would be better to migrate from YEAR(4) to YEAR without implementing sqldef.

**Comment by gfx** [comment]
Fixed in https://github.com/sqldef/sqldef/pull/1034

sqldef might say warnings in the future, though.
